File: julia/triangle/triangle.jl
is_equilateral(sides) = is_triangle(sides) && sides[1] == sides[2] == sides[3]
is_isosceles(sides) = is_triangle(sides) && sides[1] == sides[2] || sides[1] == sides[3] || sides[2] == sides[3]
is_scalene(sides) = is_triangle(sides) && sides[1] != sides[2] && sides[1] != sides[3] && sides[2] != sides[3]
is_triangle(sides) = all(s -> s!=0, sides) && sides[1] + sides[2] >= sides[3] && sides[1] + sides[3] >= sides[2] && sides[2] + sides[3] >= sides[1]

File: csharp/triangle/Triangle.cs
using System;
using System.Linq;

public static class Triangle
{
    public static bool IsScalene(double side1, double side2, double side3)
    {
        var sides = new[] { side1, side2, side3 };
        return IsTriangle(sides) && sides.All(s => sides.Where(ss => ss == s).Count() == 1);
    }

    public static bool IsIsosceles(double side1, double side2, double side3)
    {
        var sides = new[] { side1, side2, side3 };
        return IsTriangle(sides) && sides.Where(s => sides.Where(ss => ss == s).Count() > 1).Count() > 1;
    }

    public static bool IsEquilateral(double side1, double side2, double side3)
    {
        var sides = new[] { side1, side2, side3 };
        return IsTriangle(sides) && sides.All(s => s == side1);
    }

    private static bool IsTriangle(double[] sides) => !sides.Any(s => s == 0 || sides.Sum() - s < s);
}